tools/http-test-helper/src/main/java/org/enso/shttp/cloud_mock/PathResolver.java

```diff
@@ -27,8 +26,19 @@ public boolean canHandle(String subPath) {
 
   @Override
   public void handleCloudAPI(CloudExchange exchange) throws IOException {
-    JsonNode root = jsonMapper.readTree(exchange.decodeBodyAsText());
-    String path = root.get("path").asText();
+    String queryString = exchange.getHttpExchange().getRequestURI().getQuery();
+    if (queryString == null) {
+      exchange.sendResponse(400, "Missing `path` parameter in query string (empty).");
+      return;
+    }
+
+    String prefix = "path=";
+    if (!queryString.startsWith(prefix)) {
+      exchange.sendResponse(400, "Missing `path` parameter in query string: `" + queryString + "`");
+      return;
+    }
+
+    String path = queryString.substring(prefix.length());
```

Comment:
Don't you still need to URI-decode the path at this point?

Reply:
Java's `URI::getQuery` returns an already decoded query fragment. `getRawQuery` is undecoded.